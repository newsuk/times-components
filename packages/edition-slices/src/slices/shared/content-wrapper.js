/**
 * ContentWrapper restricts the content area of each slice. It is used together with the Gutter component,
 * which sets the gutter space around the content. This component is currently used only for huge breakpoints.
 */

import React from "react";
import PropTypes from "prop-types";
import { TcView } from "@times-components/utils";
import styleFactory from "./styles";

const styles = styleFactory();

const ContentWrapper = ({ children }) => (
  <TcView style={styles.contentWrapperStyles}>{children}</TcView>
);

ContentWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default ContentWrapper;
