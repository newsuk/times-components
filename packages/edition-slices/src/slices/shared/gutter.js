import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import { editionMaxWidth } from "@times-components/styleguide";

const Gutter = ({ children }) => (
  <View
    style={{
      alignSelf: "center",
      maxWidth: "100%",
      width: editionMaxWidth
    }}
  >
    {children}
  </View>
);

Gutter.propTypes = {
  children: PropTypes.node.isRequired
};

export default Gutter;
