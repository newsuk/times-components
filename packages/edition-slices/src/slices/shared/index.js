import Gutter from "./gutter";
import ContentWrapper from "./content-wrapper";
import ResponsiveSlice from "./responsive-slice";

export { Gutter, ContentWrapper, ResponsiveSlice };
