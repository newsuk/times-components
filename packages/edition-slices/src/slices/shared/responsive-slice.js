import React from "react";
import PropTypes from "prop-types";
import { ResponsiveContext } from "@times-components/responsive";
import { editionBreakpoints } from "@times-components/ts-styleguide";
import Gutter from "./gutter";
import ContentWrapper from "./content-wrapper";

const ResponsiveSlice = ({
  renderSmall,
  renderMedium,
  renderWide,
  renderHuge
}) => (
  <ResponsiveContext.Consumer>
    {({ editionBreakpoint }) => {
      switch (editionBreakpoint) {
        case editionBreakpoints.small:
          return renderSmall(editionBreakpoint);
        case editionBreakpoints.medium:
          return <Gutter>{renderMedium(editionBreakpoint)}</Gutter>;
        case editionBreakpoints.wide:
          return (
            <Gutter>
              {(renderWide && renderWide(editionBreakpoint)) ||
                (renderMedium && renderMedium(editionBreakpoint))}
            </Gutter>
          );
        case editionBreakpoints.huge:
          return (
            <Gutter>
              <ContentWrapper>
                {(renderHuge && renderHuge(editionBreakpoint)) ||
                  (renderWide && renderWide(editionBreakpoint)) ||
                  (renderMedium && renderMedium(editionBreakpoint))}
              </ContentWrapper>
            </Gutter>
          );
        default:
          return renderSmall(editionBreakpoint);
      }
    }}
  </ResponsiveContext.Consumer>
);

ResponsiveSlice.propTypes = {
  renderSmall: PropTypes.func.isRequired,
  renderMedium: PropTypes.func.isRequired,
  renderWide: PropTypes.func,
  renderHuge: PropTypes.func
};

ResponsiveSlice.defaultProps = {
  renderWide: null,
  renderHuge: null
};

export default ResponsiveSlice;
