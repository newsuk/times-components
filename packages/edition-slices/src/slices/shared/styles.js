import {
  editionMaxWidth,
  sliceContentMaxWidth,
  spacing,
  editionBreakpoints
} from "@times-components/ts-styleguide";

const tabletPaddingHorizontalResolver = {
  [editionBreakpoints.medium]: spacing(2),
  [editionBreakpoints.wide]: spacing(6),
  [editionBreakpoints.huge]: spacing(2)
};

export default (isTablet, breakpoint) => ({
  contentWrapperStyles: {
    flex: 1,
    alignSelf: "center",
    width: sliceContentMaxWidth
  },

  gutterStyles: {
    alignSelf: "center",
    maxWidth: "100%",
    paddingHorizontal: isTablet
      ? tabletPaddingHorizontalResolver[breakpoint]
      : 0,
    width: editionMaxWidth
  }
});
