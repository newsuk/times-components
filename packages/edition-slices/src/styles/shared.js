const sharedStyle = {
  EditionSlicesBody: {
    padding: 20
  }
};

export default sharedStyle;
