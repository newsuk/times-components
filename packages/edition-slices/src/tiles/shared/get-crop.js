export default (leadAsset, crop) => {
  if (!leadAsset) return null;

  // eslint-disable-next-line no-underscore-dangle
  const isVideo = leadAsset.__typename === "Video";
  const displayImage = isVideo
    ? leadAsset.posterImage[crop] || {}
    : leadAsset[crop] || {};

  return displayImage.url;
};
