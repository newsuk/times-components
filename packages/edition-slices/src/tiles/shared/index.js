import getCrop from "./get-crop";
import TileImage from "./tile-image";
import TileLink from "./tile-link";
import TileSummary from "./tile-summary";
import withTileTracking from "./tile-tracking-events";
import withPuzzleTileTracking from "./puzzle-tracking-events";

export {
  getCrop,
  TileImage,
  TileLink,
  TileSummary,
  withPuzzleTileTracking,
  withTileTracking
};
