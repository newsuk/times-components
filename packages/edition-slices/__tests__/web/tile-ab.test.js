import shared from "../tile-ab/shared-tile-ab.web";

shared();
