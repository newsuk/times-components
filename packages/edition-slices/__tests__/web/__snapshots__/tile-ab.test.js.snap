// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tile ab huge 1`] = `
<style>
.IS1 {
  margin-bottom: 0px;
}

.IS2 {
  color: #333333;
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5;
  font-size: 45;
  line-height: 45px;
}

.IS3 {
  color: inherit;
  font: inherit;
  white-space: inherit;
}

.IS4 {
  color: #696969;
  flex-wrap: wrap;
  margin-bottom: 10px;
  font-family: TimesDigitalW04;
  font-size: 14;
  line-height: 20px;
  margin-top: 5;
}

.IS5 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS6 {
  -webkit-flex-grow: 0.74;
  flex-grow: 0.74;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  padding-right: 20px;
  -ms-flex-positive: 0.74;
  -webkit-box-flex: 0.74;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS7 {
  flex: 0.26;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS6"
  >
    <div
      className="css-view-1dbjc4n IS5"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="tc-view__TcView-nuazoi-0 fPjBcr"
        >
          <div
            className="tc-view__TcView-nuazoi-0 fPjBcr IS1"
          >
            <ArticleLabel
              color="#005B8D"
              isVideo={false}
              title="label"
            />
          </div>
          <div
            aria-level="3"
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS2"
            role="heading"
          >
            This is tile headline
          </div>
          <div
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS4"
          >
            <div
              className="tc-text__TcText-sc-15igzev-0 lhcAkf IS3"
            >
              
              Senior ministers accused Labour today of trying to “frustrate and cancel Brexit” after the party announced plans to hijack Boris Johnson’s deal with amendments for a second referendum and customs union with the EU. Rishi Sunak, the chief secretary to the Treasury, warned that any attempt by MPs to add a customs union to the government’s Brexit deal would be “procedural tricks” intended to frustrate departure. His comments came as ministers announced that they planned to publish the legislation that would formally take Britain out of the European Union today. They will also apply to the Speaker to hold another so-called meaningful vote on the deal following Mr Johnson’s decision to comply with the Benn act, which forces the prime minister to request a Brexit
              ...
            </div>
          </div>
          <ArticleFlags
            flags={
              Array [
                Object {
                  "expiryTime": "2030-03-14T12:00:00.000Z",
                  "type": "EXCLUSIVE",
                },
              ]
            }
            longRead={true}
          />
        </div>
      </div>
    </div>
  </div>
  <Image
    aspectRatio={0.6666666666666666}
    className="IS7"
    fill={true}
    uri="//www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F5a14c64e-06db-11e9-abe2-4909b2eb0130.jpg?crop=1195%2C1792%2C747%2C0"
  />
</Link>
`;

exports[`tile ab medium 1`] = `
<style>
.IS1 {
  margin-bottom: 0px;
}

.IS2 {
  color: #333333;
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5;
  font-size: 30;
  line-height: 30px;
}

.IS3 {
  color: inherit;
  font: inherit;
  white-space: inherit;
}

.IS4 {
  color: #696969;
  flex-wrap: wrap;
  margin-bottom: 10px;
  font-family: TimesDigitalW04;
  font-size: 14;
  line-height: 20px;
  margin-top: 5;
}

.IS5 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS6 {
  -webkit-flex-grow: 0.56;
  flex-grow: 0.56;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  padding-right: 20px;
  -ms-flex-positive: 0.56;
  -webkit-box-flex: 0.56;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS7 {
  flex: 0.44;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS6"
  >
    <div
      className="css-view-1dbjc4n IS5"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="tc-view__TcView-nuazoi-0 fPjBcr"
        >
          <div
            className="tc-view__TcView-nuazoi-0 fPjBcr IS1"
          >
            <ArticleLabel
              color="#005B8D"
              isVideo={false}
              title="label"
            />
          </div>
          <div
            aria-level="3"
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS2"
            role="heading"
          >
            This is tile headline
          </div>
          <div
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS4"
          >
            <div
              className="tc-text__TcText-sc-15igzev-0 lhcAkf IS3"
            >
              
              Senior ministers accused Labour today of trying to “frustrate and cancel Brexit” after the party announced plans to hijack Boris Johnson’s deal with amendments for a second referendum and customs union with the EU. Rishi Sunak, the chief secretary to the Treasury, warned that any attempt by MPs to add a customs union to the government’s Brexit deal would be “procedural tricks” intended to frustrate departure. His comments came as ministers announced that they planned to publish the legislation that would formally take Britain out of the European Union today. They will also apply to the Speaker to hold another so-called meaningful vote on the deal following Mr Johnson’s decision to comply with the Benn act, which forces the prime minister to request a Brexit
              ...
            </div>
          </div>
          <ArticleFlags
            flags={
              Array [
                Object {
                  "expiryTime": "2030-03-14T12:00:00.000Z",
                  "type": "EXCLUSIVE",
                },
              ]
            }
            longRead={true}
          />
        </div>
      </div>
    </div>
  </div>
  <Image
    aspectRatio={0.6666666666666666}
    className="IS7"
    fill={true}
    uri="//www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F5a14c64e-06db-11e9-abe2-4909b2eb0130.jpg?crop=1195%2C1792%2C747%2C0"
  />
</Link>
`;

exports[`tile ab wide 1`] = `
<style>
.IS1 {
  margin-bottom: 0px;
}

.IS2 {
  color: #333333;
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5;
  font-size: 40;
  line-height: 40px;
}

.IS3 {
  color: inherit;
  font: inherit;
  white-space: inherit;
}

.IS4 {
  color: #696969;
  flex-wrap: wrap;
  margin-bottom: 10px;
  font-family: TimesDigitalW04;
  font-size: 14;
  line-height: 20px;
  margin-top: 5;
}

.IS5 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS6 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  padding-right: 20px;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS7 {
  flex: 1;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS6"
  >
    <div
      className="css-view-1dbjc4n IS5"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="tc-view__TcView-nuazoi-0 fPjBcr"
        >
          <div
            className="tc-view__TcView-nuazoi-0 fPjBcr IS1"
          >
            <ArticleLabel
              color="#005B8D"
              isVideo={false}
              title="label"
            />
          </div>
          <div
            aria-level="3"
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS2"
            role="heading"
          >
            This is tile headline
          </div>
          <div
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS4"
          >
            <div
              className="tc-text__TcText-sc-15igzev-0 lhcAkf IS3"
            >
              
              Senior ministers accused Labour today of trying to “frustrate and cancel Brexit” after the party announced plans to hijack Boris Johnson’s deal with amendments for a second referendum and customs union with the EU. Rishi Sunak, the chief secretary to the Treasury, warned that any attempt by MPs to add a customs union to the government’s Brexit deal would be “procedural tricks” intended to frustrate departure. His comments came as ministers announced that they planned to publish the legislation that would formally take Britain out of the European Union today. They will also apply to the Speaker to hold another so-called meaningful vote on the deal following Mr Johnson’s decision to comply with the Benn act, which forces the prime minister to request a Brexit
              ...
            </div>
          </div>
          <ArticleFlags
            flags={
              Array [
                Object {
                  "expiryTime": "2030-03-14T12:00:00.000Z",
                  "type": "EXCLUSIVE",
                },
              ]
            }
            longRead={true}
          />
        </div>
      </div>
    </div>
  </div>
  <Image
    aspectRatio={0.6666666666666666}
    className="IS7"
    fill={true}
    uri="//www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F5a14c64e-06db-11e9-abe2-4909b2eb0130.jpg?crop=1195%2C1792%2C747%2C0"
  />
</Link>
`;

exports[`tile ab without breakpoint should be like medium 1`] = `
<style>
.IS1 {
  margin-bottom: 0px;
}

.IS2 {
  color: #333333;
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5;
  font-size: 30;
  line-height: 30px;
}

.IS3 {
  color: inherit;
  font: inherit;
  white-space: inherit;
}

.IS4 {
  color: #696969;
  flex-wrap: wrap;
  margin-bottom: 10px;
  font-family: TimesDigitalW04;
  font-size: 14;
  line-height: 20px;
  margin-top: 5;
}

.IS5 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS6 {
  -webkit-flex-grow: 0.56;
  flex-grow: 0.56;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  padding-right: 20px;
  -ms-flex-positive: 0.56;
  -webkit-box-flex: 0.56;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}

.IS7 {
  flex: 0.44;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS6"
  >
    <div
      className="css-view-1dbjc4n IS5"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="tc-view__TcView-nuazoi-0 fPjBcr"
        >
          <div
            className="tc-view__TcView-nuazoi-0 fPjBcr IS1"
          >
            <ArticleLabel
              color="#005B8D"
              isVideo={false}
              title="label"
            />
          </div>
          <div
            aria-level="3"
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS2"
            role="heading"
          >
            This is tile headline
          </div>
          <div
            className="tc-text__TcText-sc-15igzev-0 lhcAkf IS4"
          >
            <div
              className="tc-text__TcText-sc-15igzev-0 lhcAkf IS3"
            >
              
              Senior ministers accused Labour today of trying to “frustrate and cancel Brexit” after the party announced plans to hijack Boris Johnson’s deal with amendments for a second referendum and customs union with the EU. Rishi Sunak, the chief secretary to the Treasury, warned that any attempt by MPs to add a customs union to the government’s Brexit deal would be “procedural tricks” intended to frustrate departure. His comments came as ministers announced that they planned to publish the legislation that would formally take Britain out of the European Union today. They will also apply to the Speaker to hold another so-called meaningful vote on the deal following Mr Johnson’s decision to comply with the Benn act, which forces the prime minister to request a Brexit
              ...
            </div>
          </div>
          <ArticleFlags
            flags={
              Array [
                Object {
                  "expiryTime": "2030-03-14T12:00:00.000Z",
                  "type": "EXCLUSIVE",
                },
              ]
            }
            longRead={true}
          />
        </div>
      </div>
    </div>
  </div>
  <Image
    aspectRatio={0.6666666666666666}
    className="IS7"
    fill={true}
    uri="//www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F5a14c64e-06db-11e9-abe2-4909b2eb0130.jpg?crop=1195%2C1792%2C747%2C0"
  />
</Link>
`;
